### ISSUE TITLE
[Feature]: Shutdown signal sometimes ignored in tx-coordinator event loop

### ISSUE THREAD

**Issue by djordon** [stacks-network/sbtc#998]
# Feature - Shutdown signal sometimes ignored in tx-coordinator event loop

## 1. Description

The coordinator sometimes creates a filtered stream of all messages that excludes shutdown notifications. These shouldn't necessarily be excluded, but instead should trigger some termination condition.

### 1.1 Context & Purpose

This should only affects whether the stream gets shutdown.

### 2.1 Acceptance Criteria:
   - [ ] If an event loop receives a shutdown notification, it should stop what it's doing and try to gracefully shutdown.

**Comment by cylewitruk** [comment]
Another option would be to allow these loops to continue and terminate gracefully, and implement the "double ctrl-c" pattern, i.e. exit ungracefully if ctrl-c is pressed a 2nd time.

**Comment by djordon** [comment]
> Another option would be to allow these loops to continue and terminate gracefully, and implement the "double ctrl-c" pattern, i.e. exit ungracefully if ctrl-c is pressed a 2nd time.

Oh hmmm. I see. So like, if we hit ctrl-c when the signer is leading a signing round, the default is to let it finish (which could take some time). But if we hit ctrl-c again the binary "brutal kills" itself?

**Comment by cylewitruk** [comment]
> > Another option would be to allow these loops to continue and terminate gracefully, and implement the "double ctrl-c" pattern, i.e. exit ungracefully if ctrl-c is pressed a 2nd time.
> 
> Oh hmmm. I see. So like, if we hit ctrl-c when the signer is leading a signing round, the default is to let it finish (which could take some time). But if we hit ctrl-c again the binary "brutal kills" itself? 

Yeah exactly, that's a pretty common imol and wouldn't tske that much to impl for us i don't think
